internal/gcsx/random_reader.go

```diff
@@ -516,26 +516,58 @@ func (rr *randomReader) startRead(start int64, end int64) (err error) {
 	return
 }
 
-// getReaderInfo determines the readType and provides the range to query GCS.
-// Range here is [start, end]. End is computed using the readType, start offset
-// and size of the data the callers needs.
-func (rr *randomReader) getReadInfo(
-	start int64,
-	size int64) (end int64, err error) {
-	// Make sure start and size are legal.
-	if start < 0 || uint64(start) > rr.object.Size || size < 0 {
-		err = fmt.Errorf(
-			"range [%d, %d) is illegal for %d-byte object",
-			start,
-			start+size,
-			rr.object.Size)
-		return
+// isSeekNeeded determines if the current read at `offset` should be considered a
+// seek, given the previous read pattern & the expected offset.
+func isSeekNeeded(readType, offset, expectedOffset int64) bool {
```

Comment:
```suggestion
func isSeekNeeded(readType, offset, expectedOffset int64) bool {
	if expectedOffset == 0 {
		return false
	}
	
	if readType == metrics.ReadTypeRandom {
		return offset != expectedOffset
	}
	
	if readType == metrics.ReadTypeSequential {
		return offset < expectedOffset || offset > expectedOffset+maxReadSize
	}
```

Ref: https://github.com/GoogleCloudPlatform/gcsfuse/pull/3577

Reply:
Done